src/main/java/com/codegym/jrugotom5/controller/AdvertController.java

```diff
@@ -39,4 +42,28 @@ public List<AdvertBasicInfoDTO> getByCategory(@PathVariable String category) {
         return advertService.getByCategory(category);
 
     }
+
+    @DeleteMapping("/advert/{id}")
+    public void deleteAdvert(@PathVariable Long id) {
+        advertService.deleteAdvertById(id);
+    }
+
+    @Transactional
```

Comment:
`@Transactional` in controller is a bad practice. [more info](https://stackoverflow.com/questions/23118789/why-we-shouldnt-make-a-spring-mvc-controller-transactional#:~:text=TLDR%3A%20this%20is%20because%20only,the%20scope%20of%20a%20transaction)

Reply:
@Transactional moved to Service